OpenDreamRuntime/Objects/MetaObjects/DreamMetaObjectWorld.cs

```diff
@@ -32,9 +32,15 @@ public DreamMetaObjectWorld(DreamRuntime runtime)
             } else if (variableName == "maxz") {
                 int newMaxZ = variableValue.GetValueAsInteger();
 
-                if (newMaxZ < oldVariableValue.GetValueAsInteger()) throw new NotImplementedException("Cannot set maxz lower than previous value");
-
-                while (Runtime.Map.Levels.Count < newMaxZ) Runtime.Map.AddLevel();
+                if (newMaxZ < oldVariableValue.GetValueAsInteger()) {
```

Comment:
Use `Runtime.Map.Levels.Count` instead of `oldVariableValue`
```suggestion
                if (newMaxZ < Runtime.Map.Levels.Count) {
```